src/sync.coffee

```diff
@@ -214,10 +214,16 @@ class Sync
             else
                 @pouch.getPreviousRev doc._id, rev, (err, old) ->
                     log.debug old
-                    if err or old.checksum isnt doc.checksum
+                    if err
                         side.overwriteFile doc, old, callback
-                    else
+                    else if old.checksum is doc.checksum
+                        side.updateFileMetadata doc, old, callback
+                    else if old.remote and not old.checksum
+                        # Photos uploaded by cozy-mobile have no checksum,
```

Comment:
I create an issue in mobile project: https://github.com/cozy/cozy-mobile/issues/227